test/e2e/ctx/ctx.go

```diff
@@ -0,0 +1,97 @@
+package ctx
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	pversioned "github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/client/clientset/versioned"
+)
+
+var ctx TestContext
+
+// TestContext represents the environment of an executing test. It can
+// be considered roughly analogous to a kubeconfig context.
+type TestContext struct {
```

Comment:
Most of this (context, provisioner, installer) is useful outside of OLM. I suggest that at some point we remove the OLM-specific imports here, move the context to its own repo or `pkg/lib`, and create an OLM-specific types that embed/decorate them when necessary.

Reply:
:+1: Agreed! I'm treating this context implementation as a rough draft (just enough to be able to point different test nodes to different clusters without making the test aware of it). Thanks for taking the time to discuss this yesterday.